src/classes/Database.php

```diff
@@ -420,6 +470,134 @@ public function deleteMonitor(string $uuid, int $userId): bool {
         }
     }
 
+    public function getMonitorIdByUuid(string $uuid): int|false {
+        $pdo = $this->getPdo();
+        try {
+            $stmt = $pdo->prepare("SELECT id FROM monitors WHERE uuid = ?");
+            $stmt->execute([$uuid]);
+            $id = $stmt->fetchColumn();
+            if ($id === false || $id === null) {
+                return false;
+            }
+            return (int)$id;
+        } catch (\PDOException $e) {
+            Logger::error("Error getting monitor id by uuid", ['uuid' => $uuid, 'error' => $e->getMessage()]);
+            throw $e;
+        }
+    }
+
+    public function startPingTracking(string $uuid): bool {
+        $pdo = $this->getPdo();
+        $monitorId = $this->getMonitorIdByUuid($uuid);
+        if ($monitorId === false) {
+            return false;
+        }
+        try {
+            $pdo->prepare("DELETE FROM ping_tracking WHERE monitor_id = ?")->execute([$monitorId]);
+            $stmt = $pdo->prepare("INSERT INTO ping_tracking (monitor_id, started_at) VALUES (?, CURRENT_TIMESTAMP)");
+            return $stmt->execute([$monitorId]);
+        } catch (\PDOException $e) {
+            Logger::error("Error starting ping tracking", ['uuid' => $uuid, 'error' => $e->getMessage()]);
+            throw $e;
+        }
+    }
+
+    public function completePing(string $uuid): array|false {
+        $pdo = $this->getPdo();
+        $monitorId = $this->getMonitorIdByUuid($uuid);
+        if ($monitorId === false) {
+            return false;
+        }
+
+        try {
+            $pdo->beginTransaction();
+
+            $stmtStart = $pdo->prepare("SELECT started_at FROM ping_tracking WHERE monitor_id = ?");
+            $stmtStart->execute([$monitorId]);
+            $startedAt = $stmtStart->fetchColumn();
+
+            $durationMs = null;
+            if ($startedAt !== false && $startedAt !== null) {
+                $tz = new \DateTimeZone('UTC');
+                $now = new \DateTimeImmutable('now', $tz);
+                $start = \DateTimeImmutable::createFromFormat('Y-m-d H:i:s', (string)$startedAt, $tz);
+                if ($start === false) {
+                    $start = new \DateTimeImmutable((string)$startedAt, $tz);
+                }
+                $diffSeconds = max(0, $now->getTimestamp() - $start->getTimestamp());
+                $durationMs = $diffSeconds * 1000;
+            }
```

Comment:
@jetbrains-junie you're losing millisecond information here. it's always calculating full seconds. fix that